### PR TITLE
api: add vim.version

### PR DESCRIPTION
**Pull request by mjlbach**
@ckipp01 @teto 

`:lua vim.version()`

v0.5.0

### CODE REVIEW

**Comment by clason** [comment]
Would it make sense to add an option to control the verbosity? (So you'd get a dict with git hash, release/debug, platform, etc. -- not sure how far it makes sense to push this...)

**Comment by ckipp01** [comment]
Just to confirm, in this scenario does the `NVIM_VERSION_PATCH` include the snapshot info as well? Like will this displaying `0.5.0` or `0.5.0-djfkladnvck` if I am using head?

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -622,6 +622,10 @@ vim.api.{func}({...})					*vim.api*
         Example: call the "nvim_get_current_line()" API function: >
             print(tostring(vim.api.nvim_get_current_line()))
 
+vim.version.()    					*vim.version*
```
Comment (clason) [inline runtime/doc/lua.txt:625]: shouldn't this be `vim.nvim_version`? (also stray period at the end)

**Comment by mjlbach** [comment]
> Just to confirm, in this scenario does the NVIM_VERSION_PATCH include the snapshot info as well? Like will this displaying 0.5.0 or 0.5.0-djfkladnvck if I am using head?

0.5.0 (patch not commit!)

> Would it make sense to add an option to control the verbosity? (So you'd get a dict with git hash, release/debug, platform, etc. -- not sure how far it makes sense to push this...)

Sure, I can add any/all of that.

File: runtime/doc/lua.txt (hunk 1 above)
Comment (mjlbach) [reply runtime/doc/lua.txt]: Fixed!

**Comment by bfredl** [comment]
> Would it make sense to add an option to control the verbosity? (So you'd get a dict with git hash, release/debug, platform, etc. -- not sure how far it makes sense to push this...)

We can also add more info to the `vim.fn.api_info().version` dict.

**Comment by teto** [comment]
my concern with api_info() was that it dumped too much. Rather than adding a new version function, maybe  we could add a filter as to what to drop in api_info (as well as adding NVIM_VERSION_PATCH etc). Something like `api_info({'version': true})` ?

**Comment by bfredl** [comment]
Or ju

**Comment by bfredl** [comment]
You can add `vim.version`, my point if you add new information it should __also__ be added to structured metadata.

**Comment by teto** [comment]
let's have `version` returns `version_dict` so that it remains in sync with api_info. anything added there will be added to both.

**Comment by mjlbach** [comment]
I changed it to return version_dict():

```lua
:lua print(vim.inspect(vim.version()))

{
  api_compatible = 0,
  api_level = 7,
  api_prerelease = true,
  major = 0,
  minor = 5,
  patch = 0
}
```

Is the naming of the func/placement/docs ok?

**Review by teto — APPROVED** [review]

**Pull request merged**